lib/analyzer/binary-version-extractors/openjdk-jre.ts

```diff
@@ -7,16 +7,18 @@ export {
 };
 
 async function extract(targetImage: string): Promise<Binary | null> {
+  let binaryVersion: string;
   try {
-    const binaryVersion = await new Docker(targetImage).
-      run('java', [ '-version' ]);
-    return parseOpenJDKBinary(binaryVersion);
-  } catch (stderr) {
+    binaryVersion = (await new Docker(targetImage).
+      run('java', [ '-version' ])).stdout;
+  } catch (error) {
+    const stderr: string = error.stderr;
     if (typeof stderr === 'string' && stderr.indexOf('not found') >= 0) {
       return null;
     }
     throw new Error(stderr);
   }
+  return parseOpenJDKBinary(binaryVersion);
```

Comment:
why did you decided to take this out of the `try {}` block?

Reply:
The try block is intended for the subprocess call, but let's leave it out as it's an unrelated change